src/main/scala/com/rbmhtechnology/calliope/scaladsl/EventStore.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2015 - 2017 Red Bull Media House GmbH <http://www.redbullmediahouse.com> - all rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rbmhtechnology.calliope.scaladsl
+
+import akka.actor.ActorSystem
+import com.rbmhtechnology.calliope._
+
+import scala.collection.immutable.Seq
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.FiniteDuration
+
+trait EventStore[A] {
+
+  def readEvents(fromSequenceNr: Long, limit: Int): Seq[StoredEvent]
+
+  def writeEvent(event: Array[Byte], topic: String, aggregateId: String, onCommit: Unit => Unit): Unit
```

Comment:
Initially I thought so too, but how would I invoke this then in the implementation of a store and wouldn't it only be invoked once as the evaluation is only delayed until the first access of `onCommit`?